spec/lib/dfe_signin/dfe_signin_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+
+RSpec.describe DfeSignin do
+  describe "#configure" do
+    before do
+      DfeSignin.configure do |config|
+        config.client_id = 123
+        config.secret = "sekrit"
+        config.base_url = "http://example.com"
+      end
+    end
+
+    after do
+      require Rails.root.join("config", "initializers", "dfe_signin")
```

Comment:
I don't think this will do what you expect - `require` won't re-evaluate the config as it tracks the files that have been required before and won't require the same file twice. If you want to re-execute the code in a file you'll want to use `load` instead.

How about instead of overriding/resetting the config in this spec, we test that the config matches the values from the environment variables the initializer file sets them to?